nisystemlink/clients/spec/utilities/_dataframe_utilities.py

```diff
@@ -0,0 +1,319 @@
+    StringConditionValue,
+)
+from nisystemlink.clients.spec.models._query_specs import (
+    SpecificationProjection,
+)
+from nisystemlink.clients.spec.models._specification import Specification
+from nisystemlink.clients.spec.utilities._client_utilities import __batch_query_specs
+from nisystemlink.clients.spec.utilities._constants import (
+    CONDITION_COLUMN_HEADER_PREFIX,
+    KEYWORDS_COLUMN_HEADER,
+    PROPERTY_COLUMN_HEADER_PREFIX,
+)
+
+
+def __serialize_conditions(conditions: List[Condition]) -> Dict[str, str]:
+    """Serialize conditions into desired format.
+
+    Args:
+        conditions: List of all conditions in a spec.
+
+    Returns:
+        Conditions as a dictionary.
+    """
+    return {
+        __generate_condition_column_header(condition): ", ".join(
+            __get_condition_values(condition)
+        )
+        for condition in conditions
+    }
+
+
+def get_specs_dataframe(
+    client: SpecClient,
+    product_id: str,
+    filter: Optional[str] = None,
+    column_projection: Optional[List[SpecificationProjection]] = None,
+    condition_format: Callable[
+        [List[Condition]], Dict[str, str]
+    ] = __serialize_conditions,
+) -> pd.DataFrame:
+    """Get specs of a specific product as a dataframe.
+
+    Args:
+        client: The Spec Client to use for the request.
+        product_ids: ID of the product to query specs.
+        filter: The specification query filter in Dynamic Linq format.
+        column_projection: List of columns to be included to the spec dataframe.
+                           This is an optional parameter. By default all the values will be retrieved.
+        condition_format: Function which takes in a list of condition objects and returns
+                          a dictionary of condition and its values. The dictionary keys
+                          should be the condition name and the values should be the condition
+                          value in any format you need. Keys will be used as the dataframe
+                          column header and values will be used as the row cells for the
+                          respective column header. For all the condition columns to be grouped
+                          together in the dataframe, the dictionary key should have the prefix "condition_".
+                          This is an optional parameter. By default column header will be
+                          "condition_<conditionName>(<conditionUnit>)".
+                          The column value will be "[min: num; max: num, step: num], num, num"
+                          where data within the '[]' is numeric condition range and other num
+                          values are numeric condition discrete values.
+                          The column value will be "str, str, str" - where str values are the
+                          condition discrete values for a string condition.
+
+    Returns:
+        The list of specs of the specified product as a dataframe.
```

Comment:
The format of the dataframe should be detailed here, not just in the condition_format doc comment